app/main.py

```diff
@@ -1,6 +1,23 @@
 from typing import Callable
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cached_results_storage = {}
+
+    @functools.wraps(func)
+    def inner(*args, **kwargs) -> any:
+        cache_key = func.__name__, args, tuple(kwargs.items())
+
+        if cache_key not in cached_results_storage:
+            func_res = func(*args, **kwargs)
+            cached_results_storage[cache_key] = func_res
+
+            print("Calculating new result")
+            return func_res
```

Comment:
avoid using additional variable here, you can skip assigning `func_res` here

Reply:
Not sure how since need to write this calculation to ceched_results_storage. If I calculate it directly in return I won't be able to extract it

Reply:
You have `return cached_results_storage[cache_key]` in 21 line. It won't be a problem to do the same here